### ISSUE TITLE
Unexpected warning when using secrets in Docker Compose on Windows

### ISSUE THREAD

**Issue by chrischu** [docker/compose#8233]
## Description of the issue

I am trying to add secrets to my Windows Docker container via Docker Compose and I got it to work, but now I get a weird warning starting the container:

## Context information (for bug reports)

**Output of `docker-compose version`**
```
docker-compose version 1.28.5, build c4eb3a1f
docker-py version: 4.4.4
CPython version: 3.9.0
OpenSSL version: OpenSSL 1.1.1g  21 Apr 2020
```

**Output of `docker version`**
```
Client: Docker Engine - Community
 Cloud integration: 1.0.9
 Version:           20.10.5
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        55c4c88
 Built:             Tue Mar  2 20:14:53 2021
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.5
  API version:      1.41 (minimum version 1.24)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Tue Mar  2 20:26:56 2021
  OS/Arch:          windows/amd64
  Experimental:     false
```

**Output of `docker-compose config`**
(Make sure to add the relevant `-f` and other flags)
```
secrets:
  all:
    file: C:\Temp\ComposeSecrets\secrets
services:
  api:
    image: mcr.microsoft.com//dotnet/framework/runtime:4.8-windowsservercore-ltsc2019
    isolation: hyperv
    secrets:
    - source: all
      target: C:\secrets\
version: '3.7'
```

## Steps to reproduce the issue

1. When using the following Docker Compose file:
api.yml
```
version: "3.7"
services:
  api:
    image: api
    secrets:
      - source: all
        target: C:\secrets\ # defines where the secrets will be available in the container
secrets:
  all:
    file: ./secrets/
```
2. with the directory structure:
```
api.yml
secrets/
  secret
```
3. and running `docker-compose -f api.yml up`

### Observed result

I get the warning:
![image](https://user-images.githubusercontent.com/447251/111973905-4262d900-8aff-11eb-88c0-cd84c613f092.png)

### Expected result

I expected no warning to show up.

## Additional information

I tried various other ways to use secrets (e.g. by not specifying a directory, but the path for every secret) but I couldn't get those solutions to work because of the following error messages:
When using
```
services:
   api:
     ...
   secrets:
      - secret
secrets:
  secret:
    file: ./secrets/secret   
```
```
ERROR: for bbea49dbfb24_api  Cannot create container for service api: invalid mount config for type "bind": invalid mount path: '/run/secrets/secret'
```
and when using
```
services:
   api:
     ...
   secrets:
      - source: secret
        target: C:/secrets/secret
secrets:
  secret:
    file: ./secrets/secret   
```
```
ERROR: for bbea49dbfb24_api  Cannot create container for service api: invalid mount config for type "bind": source path must be a directory
```

**Additional environment details**
It's a physical Windows machine with version:
![image](https://user-images.githubusercontent.com/447251/111975204-9d490000-8b00-11eb-8dd8-2dd1085f8fd5.png)

**Comment by chrischu** [comment]
I moved this issue to this repository after being told that it fits better her (previous issue: https://github.com/moby/moby/issues/42180).

**Comment by drauch** [comment]
Uhm, this is still a problem. Are we in the wrong issue tracker for this?

Best regards,
D.R.

**Comment by drauch** [comment]
This is still a problem, don't mark issues as stale just because you don't react to them...

**Comment by drauch** [comment]
This is still a problem, don't mark issues as stale just because you don't react to them...

**Comment by drauch** [comment]
This is still a problem, don't mark issues as stale just because you don't react to them...

**Comment by fschmied** [comment]
Probably/maybe fixed through #10819 (#10817).
